### PR TITLE
[Owl] Add nested navigation

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
This time using a non-snapshot build. It's similar to #438 which was reverted in #445

### CODE REVIEW

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt [hunk 1]
```diff
@@ -41,23 +46,54 @@ object MainDestinations {
...
     ) {
         composable(MainDestinations.ONBOARDING_ROUTE) {
-            Onboarding(onboardingComplete = actions.onboardingComplete)
+            // Intercept back in Onboarding: make it finish the activity
+            BackHandler {
+                finishActivity()
+            }
+
+            Onboarding(
+                onboardingComplete = {
```
Comment (chrisbanes) [inline Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:74]: Trailing lambda?

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt [hunk 2]
```diff
@@ -41,23 +46,54 @@ object MainDestinations {
...
-    val navController = rememberNavController()
+fun NavGraph(
+    modifier: Modifier = Modifier,
+    finishActivity: () -> Unit = {},
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = MainDestinations.COURSES_ROUTE,
+    showOnboardingInitially: Boolean = true
+) {
+    // Onboarding could be read from shared preferences.
+    val onboardingComplete = remember(showOnboardingInitially) {
```
Comment (chrisbanes) [inline Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:57]: This code messed with my head. It's correct, but it took me a minute to work it out.

File: Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt [hunk 3]
```diff
@@ -16,21 +16,98 @@
...
 
-    CompositionLocalProvider(LocalBackDispatcher provides backDispatcher) {
-        ProvideWindowInsets {
-            ProvideImageLoader {
-                NavGraph()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.arguments?.getString(KEY_ROUTE)
+        ?: CourseTabs.FEATURED.route
+
+    if (currentRoute in CourseTabs.values().map { it.route }) {
```
Comment (chrisbanes) [inline Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt:74]: nit: extract `CourseTabs.values().map { it.route }` to  const or `remember` or something. `values()` creates a new array every call, and then so will `map`

File: Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt [hunk 4]
```diff
@@ -16,21 +16,98 @@
...
+                    unselectedContentColor = LocalContentColor.current,
+                    modifier = Modifier.navigationBarsPadding()
+                )
             }
         }
     }
 }
+
+@Composable
+fun OwlScaffold(
```
Comment (chrisbanes) [inline Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt:102]: Just use `Scaffold`? The only thing this is really doing is wrapping the content padding into a modifier.

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt (hunk 1 above)
Comment (JoseAlcerreca) [reply Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:74]: When the function of the trailing lambda is not obvious like in this case I prefer to keep the name

File: Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt (hunk 3 above)
Comment (JoseAlcerreca) [reply Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt]: Remembered the array and the map, although I suspect this is less performant overall :)

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt (hunk 2 above)
Comment (JoseAlcerreca) [reply Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:57]: Yeah... we need a better API for this

File: Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt (hunk 3 above)
Comment (chrisbanes) [reply Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt]: Remembering probably is, but storing the end map in a companion object would be optimal.

**Review by chrisbanes — APPROVED** [review]

**Comment by JoseAlcerreca** [comment]
Thanks!

**Pull request merged**